src/microbe_stage/editor/MicrobeEditorGUI.cs

```diff
@@ -1421,4 +1415,93 @@ private void OnSpeciesNameTextChanged(string newText)
             speciesNameEdit.Set("custom_colors/font_color", new Color(1, 1, 1));
         }
     }
+    private void createAndUpdateProcessBar (KeyValuePair <string, float> process, ProgressBar parent)
+    {
+        if (parent.HasNode(process.Key))
+        {
+            IconProgressBar progressBar = (IconProgressBar) parent.GetNode(process.Key);
+            TextureRect textureRect = progressBar.GetChild<TextureRect>(0);
```

Comment:
This kind of direct access of the contents of a separate scene should not be done. Instead make separate methods for the IconProgressBar class to set the min value, max value, and current value, which makes it resize itself. And also for handling the icon.